src/main/java/mate/academy/bookstore/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package mate.academy.bookstore.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.UserRegistrationRequestDto;
+import mate.academy.bookstore.dto.UserResponseDto;
+import mate.academy.bookstore.entity.User;
+import mate.academy.bookstore.exception.RegistrationException;
+import mate.academy.bookstore.mapper.UserMapper;
+import mate.academy.bookstore.repository.UserRepository;
+import mate.academy.bookstore.service.UserService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto) {
+        if (userRepository.findAllByEmail(requestDto.getEmail()).isPresent()) {

```

Comment:
a bit more efficient and simple
```suggestion
        if (userRepository.existsByEmail(requestDto.getEmail())) {
```